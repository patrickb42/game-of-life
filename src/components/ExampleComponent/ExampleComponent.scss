@import '../../scss/1-utils/#index';

.example-component {
  color: red;
  border: 1px solid blue;

  @include tablet {
    color: blue;
    border: 1px solid red;
  }
}
