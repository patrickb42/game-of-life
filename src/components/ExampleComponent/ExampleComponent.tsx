import React from 'react';
import {
  // connect,
  useSelector,
  useDispatch,
} from 'react-redux';
// import type { Record } from 'immutable';

import { exampleFunction } from '../../redux/actions';
// eslint-disable-next-line no-unused-vars
import type { ExampleReducerState } from '../../redux/reducers/exampleReducer';
// import { buildMapStateToProps } from '../../utils';

import './ExampleComponent.scss';

// interface ExampleComponentProps {
//   exampleVariable: String,
//   click: Function,
// }

// const ExampleComponent = ({ exampleVariable, click }: ExampleComponentProps) => {
const ExampleComponent = () => { // hooks
  const dispatch = useDispatch(); // hooks
  const exampleVariable = useSelector(
    (state: ExampleReducerState) => state.exampleVariable,
  ); // hooks

  return (
    <>
      <button
        type="button"
        className="example-component"
        // onClick={() => click('Example Payload')}
        onClick={() => dispatch(exampleFunction('Example Payload'))} // hooks
      >
        {`${exampleVariable} Component`}
      </button>
    </>
  );
};

export default ExampleComponent; // hooks

// const propsShape = {
//   exampleVariable: '',
// };

// export default connect(
//   buildMapStateToProps({ propsShape }),
//   { click: exampleFunction },
// )(ExampleComponent);
