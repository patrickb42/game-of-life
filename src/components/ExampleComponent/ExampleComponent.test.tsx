/* global describe it */
/* global expect */

import React from 'react';
import renderer from 'react-test-renderer';

import ExampleComponent from './ExampleComponent';

describe('<Dashboard />', () => {
  it('should match snapshot', () => {
    const component = renderer.create(<ExampleComponent />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
