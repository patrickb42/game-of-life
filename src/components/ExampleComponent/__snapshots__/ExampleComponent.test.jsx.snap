// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should match snapshot 1`] = `
<p
  className="example-component"
>
  Example Component
</p>
`;
